test/integration/test_utils.js

```diff
@@ -119,13 +130,20 @@ const waitForStorageEntries = async (page, nEntries) => {
 };
 exports.waitForStorageEntries = waitForStorageEntries;
 
-const waitForSelectedEditor = async (page, selector) => {
+const waitForSerialized = async (page, nEntries) => {
   await page.waitForFunction(
-    sel => document.querySelector(sel).classList.contains("selectedEditor"),
+    n =>
+      (window.PDFViewerApplication.pdfDocument.annotationStorage.serializable
+        ?.map?.size ?? 0) === n,
```

Comment:
Nit: Looking at [the `serializable` getter](https://github.com/mozilla/pdf.js/blob/52862893f5bef7ff6c7b8f816154d8fec00f49e7/src/display/annotation_storage.js#L178-L215) it should always return something, hence we ought to be able to skip the *first* optional chaining here.
```suggestion
      (window.PDFViewerApplication.pdfDocument.annotationStorage.serializable
        .map?.size ?? 0) === n,
```